server/svix-server/src/v1/utils.rs

```diff
@@ -24,13 +24,85 @@ const fn default_limit() -> u64 {
 }
 
 #[derive(Debug, Deserialize, Validate)]
-pub struct Pagination<T: Validate> {
+pub struct Pagination<T: Validate + Clone> {
     #[serde(default = "default_limit")]
     pub limit: u64,
     #[validate]
     pub iterator: Option<T>,
 }
 
+#[derive(Debug, Deserialize, Validate)]
+pub struct ReversablePagination<T: 'static + Validate + Clone + From<String>> {
+    #[serde(default = "default_limit")]
+    pub limit: u64,
+    #[validate]
+    pub iterator: Option<PaginationIterator<T>>,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum PaginationIterator<T: Validate + Clone> {
+    Normal(T),
+    Prev(T),
+}
+
```

Comment:
Deserialize alternative (so instead of all of the below) would be something like in https://github.com/svix/svix-webhooks/blob/main/server/svix-server/src/core/types.rs#L439

So:
1. Deserialize to string.
2. Check if it starts with `-`. If yes: serialize as `MessageId` for everything after the first char. If no: serialize as `MessageId`. You can just take the string you serialized in other case.
3. Validate should remain like you did it.